### PR TITLE
Solve printing issues in Stats with matrix expressions by adding 'is_commutative'.

### LINKED ISSUE #24132: print fails with MatMul and Expectation.
The following code worked in 1.10.1 but not in 1.11.1.

```python
from sympy import *
from sympy.stats import Expectation
from sympy.stats.rv import RandomSymbol

print(RandomSymbol("ve") * Identity(2)) # works
z = Expectation(RandomSymbol("ve")) * Identity(2) # works
print(z) # NotImplementedError

```

I have tracked down this error to the cause that
MatMul.could_extract_minus_sign exists in 1.10.1 but not in 1.11.1 by
the commit below. I don't know previous bugs of the commit.

https://github.com/sympy/sympy/commit/1d3c89501c65e4b6cd3b635be60ba1d2bf003b4d

### PR DESCRIPTION
**Pull request by faze-geek**
#### References to other Issues or PRs

Fixes #24132 

#### Brief description of what is fixed or changed
Added `is_commutative` assumption as class variable for Probability and `eval_is_commutative()` method for Expectation, Variance, Covariance classes.

```
>>> from sympy import *
>>> from sympy.stats import Expectation
>>> from sympy.stats.rv import RandomSymbol
>>> z = Expectation(RandomSymbol("x")) * Identity(2)
>>> print(z)  
NotImplementedError: noncommutative scalars in MatMul are not supported. # On master

>>> z = Expectation(RandomSymbol("x")) * Identity(2)
>>> print(z)
Expectation(x)*I  # On branch
```

#### Other comments
Initial changes to fix the issue were introduced by @KuldeepBorkar but they lacked testing and had multiple `doctest` failures. Those have been resolved here. I will add him as co-author to the commit, once this is close to merging.

#### Release Notes

* stats
  * Fixed failing print for Probability, Expectation, Variance and Covariance.

### CODE REVIEW

**Comment by faze-geek** [comment]
@sylee957 Do you mind taking a look at this.? `matrices.expressions` and `stats` labels could be added here. Thanks !

File: sympy/stats/symbolic_probability.py [hunk 1]
```diff
@@ -59,6 +59,7 @@ class Probability(Expr):
     >>> prob.evaluate_integral()
     sqrt(2)*(-sqrt(2)*sqrt(pi)*erf(sqrt(2)/2) + sqrt(2)*sqrt(pi))/(4*sqrt(pi))
     """
+    is_commutative = True
     def __new__(cls, prob, condition=None, **kwargs):
```
Comment (oscarbenjamin) [inline sympy/stats/symbolic_probability.py:63]: There should be a blank line above and below the `is_commutative` line.

**Comment by oscarbenjamin** [comment]
Are there any examples in the codebase where `Variance` or `Covariance` would be non-commutative?

File: sympy/stats/symbolic_probability.py (hunk 1 above)
Comment (faze-geek) [reply sympy/stats/symbolic_probability.py:65]: Sure !

**Comment by faze-geek** [comment]
> Are there any examples in the codebase where `Variance` or `Covariance` would be non-commutative?

Not sure of this at the moment. I will do some sleuthing soon.

**Comment by oscarbenjamin** [comment]
The reason I ask about Variance being non-commutative is that I'm not sure that the concept is well defined in the non-commutative case or if it should need some other more complicated definition. Expectation is straight-forward for something like a random matrix. We have for example:
```python
In [10]: Variance(X).rewrite(Expectation)
Out[10]: 
   2       ⎡ 2⎤
- E [X] + E⎣X ⎦
```
Notice that the derivation of this formula assumes that `X` and `E[X]` commute:
https://en.wikipedia.org/wiki/Variance#Definition

File: sympy/printing/tests/test_str.py [hunk 2]
```diff
@@ -1173,3 +1176,17 @@ def test_printing_str_array_expressions():
...
+    x = RandomSymbol("x")
+    y = RandomSymbol("y")
+    z1 = Probability(x > 0)*Identity(2)
+    z2 = Expectation(x)*Identity(2)
+    z3 = Variance(x)*Identity(2)
+    z4 = Covariance(x, y) * Identity(2)
+    assert str(z1) == "Probability(x > 0)*I"
+    assert str(z2) == "Expectation(x)*I"
+    assert str(z3) == "Variance(x)*I"
+    assert str(z4) ==  "Covariance(x, y)*I"
```
Comment (oscarbenjamin) [inline sympy/printing/tests/test_str.py:1192]: It is good to have printing tests but there should also be direct tests for `is_commutative` in the stats module.

**Pull request closed without merging**